xc7/yosys/synth.tcl

```diff
@@ -2,21 +2,29 @@ yosys -import
 
 plugin -i xdc
 plugin -i fasm
-#Import the commands from the plugins to the tcl interpreter
-yosys -import
 
-# Map (actually rename) explicitly instantiated IBUFs/OBUFs in order to
-# distinguish them from the inferred ones.
-techmap -map $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iob_map.v
+# Import the commands from the plugins to the tcl interpreter
+yosys -import
 
 # -flatten is used to ensure that the output eblif has only one module.
 # Some of symbiflow expects eblifs with only one module.
-synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp -iopad
-
-# Map inferred IBUFs/OBUFs to wires. This way ROI targets will work.
-techmap -map $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iob_make_wires.v
-# Map previously renamed IBUFs/OBUFs back to their former names.
-techmap -map $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iob_unmap.v
+#
+# Do not infer IOBs for targets that use a ROI.
+if { $::env(USE_ROI) == "TRUE" } {
+    synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp
+} else {
+    # Read Yosys baseline library first.
+    read_verilog -lib -D_EXPLICIT_CARRY +/xilinx/cells_sim.v
+    read_verilog -lib +/xilinx/cells_xtra.v
+
+    # Overwrite some models (e.g. IBUF with more parameters)
+    read_verilog -lib $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iobs.v
+
+    hierarchy -check -auto-top
+
+    # Start flow after library reading
+    synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp -iopad -run prepare:check
```

Comment:
Why does this run from `prepare` to `check` while the other pathway runs everything?

Reply:
Did you read the comment?

Reply:
Oh - I think I understand now.

The first couple of lines in this section do the library reading. Then you start from the prepare stage (which is the first stage after library reading)?

Reply:
Yes